xbmc/epg/GUIEPGGridContainer.cpp

```diff
@@ -1429,9 +1429,11 @@ bool CGUIEPGGridContainer::OnMouseWheel(char wheel, const CPoint &point)
 
 CPVRChannelPtr CGUIEPGGridContainer::GetChannel(int iIndex)
 {
+  CSingleLock lock(m_critSection);
+
   if (iIndex >= 0 && (size_t) iIndex < m_channelItems.size())
   {
-    CFileItemPtr fileItem = m_channelItems[iIndex];
+    const CFileItemPtr &fileItem = m_channelItems[iIndex];
```

Comment:
as there's no copy done the ref count in the shared_ptr isn't incremented, if another thread erases the vector the ref count might drop to 0 and the item is destroyed leaving a reference pointing into random memory